docs/rules/structured-imports.md

```diff
@@ -0,0 +1,33 @@
+# Enforces a certain structure in your import statements
+
+## Rationale
+- Consistency in import statement structure throughout the codebase
```

Comment:
This states a Shopify-specific end goal rather than a rationale.  Looking through other rationales, `adjacent-overload-signature` has this:

> Improves readability and organization by grouping naturally related items together.

That looks good to me.